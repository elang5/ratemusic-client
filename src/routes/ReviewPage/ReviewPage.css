.review-page {
  display: flex;
  flex-flow: column wrap;
  align-items: flex-start;
  align-content: center;
  padding-left: 2rem;
  margin-top: 3rem;
  max-width: 85%;
  padding-bottom: 1rem;
}

.title {
  margin: 1rem 0;
}

.review-page > p {
  font-size: 18px;
}

.album-art {
  padding-top: 4rem;
}

.review-content {
  margin-bottom: 2em;
}

@media only screen and (max-width: 575px) {
  .album-art {
    width: 60%
  }
}