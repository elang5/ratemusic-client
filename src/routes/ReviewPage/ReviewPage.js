import React, { Component } from 'react'
import AlbumsApiService from '../../services/albums-api-service'
import './ReviewPage.css'

export class ReviewPage extends Component {
  constructor(props) {
    super(props)

    this.state = {
      review: [],
      album_art: '',
      album_name: ''
    }
  }

  componentDidMount() {
    const { albumId, reviewId } = this.props.match.params
    AlbumsApiService.getAlbumReview(albumId, reviewId)
      .then(res => {
        this.setState({ review: res[0] })
      })
      .catch(err => this.setState({ error: err.error }))

    AlbumsApiService.getAlbum(this.props.match.params.albumId)
      .then(res => {
        this.setState({ 
          album_art: res.images[1].url,
          album_name: res.name })
      })
      .catch(err => this.setState({ error: err.error }))
  }

  render() {
    const { album_art, album_name, review } = this.state
    return (
      <div className="review-page">
        <img src={album_art} alt={album_name} className="album-art" />
        <h2 className="title">{`${review.title}`}</h2>
        <p>{`By: ${review.user_name}`}</p>
        <p>{`Rating: ${review.rating}`}</p>
        <p className="review-content">{review.content}</p>
      </div>
    )
  }
}

export default ReviewPage
