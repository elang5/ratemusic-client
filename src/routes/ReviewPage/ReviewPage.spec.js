import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ReviewPage from './ReviewPage'

describe(`ReviewPage Component`, () => {
  const props = {
    match: {
      params: {
        albumId: 1
      }
    }
  }
  it(`renders the ReviewPage component`, () => {
    const wrapper = shallow(<ReviewPage {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})