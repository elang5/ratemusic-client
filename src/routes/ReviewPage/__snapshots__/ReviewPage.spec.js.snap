// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewPage Component renders the ReviewPage component 1`] = `
<div
  className="review-page"
>
  <a
    className="album-listen"
    href=""
    rel="noopener noreferrer"
    target="_blank"
  >
    <button
      className="listen-btn"
    >
      Listen to 
    </button>
  </a>
  <Link
    className="album-art-link"
    to="/albums/1"
  >
    <Loader
      loading={true}
    />
    <img
      alt=""
      className="album-art"
      src=""
    />
  </Link>
  <h2
    className="title"
  >
    undefined
  </h2>
  <p>
    By: undefined
  </p>
  <AlbumStarRating />
  <p
    className="review-content"
  />
</div>
`;
